projects/rocprofiler-systems/cmake/Packages.cmake

```
@@ -232,6 +232,110 @@ if(ROCPROFSYS_USE_ROCM)
     target_link_libraries(rocprofiler-systems-rocm INTERFACE amd-smi::amd-smi)
 endif()
 
+# ----------------------------------------------------------------------------------------#
+#
+# ROCpd
+#
+# ----------------------------------------------------------------------------------------#
+
+function(ROCPROFSYS_CONFIGURE_ROCPD_SCHEMA_FILES)
+    rocprofiler_systems_target_compile_definitions(
+        rocprofiler-systems-rocm INTERFACE ROCPROFSYS_USE_ROCPD_LIBRARY=0
+    )
+
+    set(SCHEMA_FILES
+        "rocpd_tables.sql"
+        "rocpd_views.sql"
+        "data_views.sql"
+        "marker_views.sql"
+        "summary_views.sql"
+    )
+
+    set(SCHEMA_SOURCE_DIR
+        "${PROJECT_SOURCE_DIR}/source/lib/core/rocpd/data_storage/schema"
+    )
+    set(SCHEMA_BINARY_DIR
+        "${PROJECT_BINARY_DIR}/source/lib/core/rocpd/data_storage/schema"
+    )
+    set(TEMPLATE_FILE "${PROJECT_SOURCE_DIR}/cmake/Templates/rocpd_schema.in")
+
+    file(MAKE_DIRECTORY ${SCHEMA_BINARY_DIR})
+
+    foreach(SCHEMA_FILE ${SCHEMA_FILES})
+        file(READ "${SCHEMA_SOURCE_DIR}/${SCHEMA_FILE}" SQL_CONTENT)
+
+        string(REPLACE "\\" "\\\\" SQL_CONTENT "${SQL_CONTENT}")
+        string(REPLACE "\"" "\\\"" SQL_CONTENT "${SQL_CONTENT}")
+        string(REPLACE "\n" "\\n\"\n\"" SQL_CONTENT "${SQL_CONTENT}")
+
+        get_filename_component(SCHEMA_NAME ${SCHEMA_FILE} NAME_WE)
+        string(TOUPPER ${SCHEMA_NAME} SCHEMA_NAME_UPPER)
+
+        configure_file("${TEMPLATE_FILE}" "${SCHEMA_BINARY_DIR}/${SCHEMA_NAME}.hpp" @ONLY)
+    endforeach()
+
+    target_include_directories(
+        rocprofiler-systems-headers
+        INTERFACE
+            $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/source/lib/core/rocpd/data_storage>
+    )
+endfunction()
+
+set(ROCPROFSYS_USE_ROCPD_LIBRARY OFF CACHE BOOL "Use rocpd library" FORCE)
+find_package(rocprofiler-sdk-rocpd ${rocprofiler_systems_FIND_QUIETLY})
+
+if(rocprofiler-sdk-rocpd_FOUND)
+    set(ROCPROFSYS_ROCPD_HAS_SQL_H FALSE)
+
+    if(rocprofiler-sdk-rocpd_DIR)
```

Could you use `INCLUDE_DIR` or `ROOT_DIR` here instead? This will save you all of the string manipulation through `get_filename_component`.

```suggestion
    if(rocprofiler-sdk-rocpd_INCLUDE_DIR)
```

For reference, 
```
-- rocprofiler-sdk-rocpd_FOUND: 1
-- rocprofiler-sdk-rocpd_DIR: /opt/rocm-7.1.0/lib/cmake/rocprofiler-sdk-rocpd
-- rocprofiler-sdk-rocpd_VERSION: 1.0.0
-- rocprofiler-sdk-rocpd_INCLUDE_DIR: /opt/rocm-7.1.0/include
-- rocprofiler-sdk-rocpd_LIBRARIES: rocprofiler-sdk-rocpd::rocprofiler-sdk-rocpd-shared-library
-- rocprofiler-sdk-rocpd_ROOT_DIR: /opt/rocm-7.1.0
-- rocprofiler-sdk-rocpd_LIB_DIR: /opt/rocm-7.1.0/lib
```